bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/item/ItemResource.java

```diff
@@ -367,6 +368,90 @@ public Response removeTag(@PathParam("itemname") String itemname, @PathParam("ta
         return Response.ok().build();
     }
 
+    /**
+     * Create or Update an item by supplying an item bean.
+     * @param itemname
+     * @param item the item bean.
+     * @return
+     */
+    @PUT
+    @Path("/{itemname: [a-zA-Z_0-9]*}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createOrUpdateItem(@PathParam("itemname") String itemname, ItemBean item) {
```

Comment:
Does this work without conflicts to the other "createOrUpdate" method, because of the different @Consumes type? I would suggest to remove the other method, as it is not needed anymore.